sound/soc/intel/boards/sof_es8336.c

```diff
@@ -119,8 +97,7 @@ static void pcm_pop_work_events(struct work_struct *work)
 
 	gpiod_set_value_cansleep(priv->gpio_speakers, priv->speaker_en);
 
-	if (quirk & SOF_ES8336_HEADPHONE_GPIO)
-		gpiod_set_value_cansleep(priv->gpio_headphone, priv->speaker_en);
+	gpiod_set_value_cansleep(priv->gpio_headphone, true);
```

Comment:
|   | Speaker GPIO |  Headphone GPIO |
| - | - | - |
| Playing music Speaker on | high | low |
| Playing music Headphone on | low | high |
| Suspend | low | low |

Reply:
When only one GPIO is used things are simple. The GPIO is high by default. Before playing music, the SPEAKER_ON dapm event is called and the GPIO set to high. When music stop, the GPIO is set to low so that there is no pop noise.

When HP GPIO is involved things become complicated. I want to do the same for HP GPIO but I don't know how. Maybe another [SND_SOC_DAPM_SUPPLY](https://elixir.bootlin.com/linux/v6.1-rc7/C/ident/SND_SOC_DAPM_SUPPLY) for headphone should be invloved or maybe the headphone status can be read by `snd_soc_dapm_get_pin_status(&card->dapm, "Headphone");` and the headphone can be turned on and off in [sof_8336_trigger](https://elixir.bootlin.com/linux/v6.1-rc7/C/ident/sof_8336_trigger)